crates/sui/src/ptb/ptb_parser/errors.rs

```diff
@@ -1,28 +1,30 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+use std::collections::BTreeMap;
+use std::fmt::Debug;
+
+use miette::{miette, LabeledSpan};
 use thiserror::Error;
 
-use super::context::FileScope;
+use crate::ptb::ptb::PTBCommand;
 
-pub type PTBResult<T> = Result<T, PTBError>;
-
-/// An error that occurred while working with a PTB. This error contains an error message along
-/// with the file scope (file name and command index) where the error occurred.
-#[derive(Debug, Clone, Error)]
-pub enum PTBError {
-    #[error("{message} at command {} in file '{}'", file_scope.file_command_index, file_scope.name)]
-    WithSource {
-        file_scope: FileScope,
-        message: String,
-    },
-}
+use super::{
+    command_token::{FILE_END, FILE_START},
+    context::FileScope,
+};
 
 #[macro_export]
 macro_rules! error {
     ($x:expr, $($arg:tt)*) => {
         return Err($crate::err!($x, $($arg)*))
     };
+    (sp: $l:expr, $x:expr, $($arg:tt)*) => {
+        return Err($crate::err!(sp: $l, $x, $($arg)*))
+    };
+    (sp: $l:expr, help: { $($h:expr),* }, $x:expr, $($arg:tt)*) => {
+        return Err($crate::err!(sp: $l, help: { $($h),* }, $x, $($arg)*))
+    };
```

Comment:
I'm a macro noob, need to spend some time to understand what these do, so cannot provide much feedback on this!

Reply:
Happy to go through them! Once you get the gist of them they're not too scary :)